pymc/logprob/binary.py

```diff
@@ -99,12 +99,10 @@ def comparison_logprob(op, values, base_rv, operand, **kwargs):
     else:
         raise TypeError(f"Unsupported scalar_op {op.scalar_op}")
 
-    if base_rv.dtype.startswith("int"):
-        logp_point = _logprob_helper(base_rv, operand, **kwargs)
-        if isinstance(op.scalar_op, GT):
-            logprob = pt.switch(condn_exp, pt.logaddexp(logprob, logp_point), logprob)
-        elif isinstance(op.scalar_op, LT):
-            logprob = pt.switch(condn_exp, logprob, pt.logaddexp(logprob, logp_point))
+    if base_rv.dtype.startswith("int") and isinstance(op.scalar_op, LT):
+        logpmf = _logprob_helper(base_rv, operand, **kwargs)
+        logcdf_lt_true = _logcdf_helper(base_rv, operand - 1, **kwargs)
+        logprob = pt.switch(condn_exp, logcdf_lt_true, pt.logaddexp(logprob, logpmf))
```

Comment:
I think this may be more readable if you define it inside the `elif(isinstance(op.scalar_op, LT)` above. You can have two branches here, for discrete and non discrete and write the two cases from scratch. the `logaddexp(logprob, logpmf)` is a bit hard to to read because the `logprob` is actually the `logcdf` but the name masks it.

What do you think?

Reply:
That's a fair point, the readability would definitely improve. I'll make the change right away!